### ISSUE TITLE
reactstrap Modal vs. React error boundaries

### ISSUE THREAD

**Issue by stefanb2** [reactstrap/reactstrap#844]
react: 16.2.0, reacstrap: 5.0.0-beta

I'm trying to add a React 16 error boundary to my application and noticed that `componentDidCatch()` in the error boundary component is not triggered for errors caused by `Modal` components. I'm following the standard example code from here: https://reactjs.org/docs/error-boundaries.html

Basic structure to show the problem

```javascript
<ErrorBoundary>
    <Container>
       ... some "normal" component here ... (A)
    </Container>
   <Modal
         isOpen={this.state.isOpen}>
     .... the children of the Modal....     (B)
   </Modal>
</ErrorBoundary>
```
If I go to some `render()` method inside A and add a `throw new Error('test');` then `ErrorBoundary::componentDidCatch()` is called. That proves that `ErrorBoundary` is working as it should.

If I go to some `render()` method inside B and add a `throw new Error('test');` then `ErrorBoundary::componentDidCatch()` is **not called**, i.e. the error is not handled just as if the error boundary wouldn't have been added. The React debug tools show the `Modal` and its children as children of `ErrorBoundary` so according to the documentation it should catch the error.

Has somebody an idea what is so special about `Modal` that makes error boundaries fail? Is it maybe because the actual DOM node for `Modal` is added parallel to the React DOM node under `<body>`? I.e. not being inside the React DOM tree would confuse React lookup code?

Any ideas?

**Comment by TheSharpieOne** [comment]
It's most likely due to reactstrap using the old `unstable_renderIntoSubtree` which was replace in react 16 with `render_portal`.
#796 changed reactstrap's Modal to use react-portal which uses react v16 and its official API for creating portals; has a fallback for react v15.
The PR mentioned above has been merged but there has not been a release since. I need to cut a release, I am hoping this solves this issue.

**Comment by stefanb2** [comment]
Thanks for the update. I'll verify this once the release is available.

For now I have found an acceptable workaround by replacing every instance of `Modal` with `ModalWithBoundary`. The drop-in replacement wraps the children into an `ErrorBoundary`:

```javascript
export default class ModalWithBoundary extends React.Component {
    render() {
        const { children, ...remainder } = this.props;
        return (
            <Modal {...remainder}>
                <ErrorBoundary children={children}/>
            </Modal>
        );
    }
}

```

**Comment by stefanb2** [comment]
reactstrap-5.0.0.beta.2 does fix the issue, but I can't upgrade due to PR #870.

**Comment by stefanb2** [comment]
Fixed by reactstrap 5.0.0-beta.3. Thanks.
